src/index.js

```diff
@@ -17,18 +18,46 @@ export function createCycleMiddleware(mainFn, drivers) {
 
         return xs.create({
           start: listener => {
-            mainListener = listener;
+            actionListener = listener;
           },
           stop: () => {},
         });
       }
 
-      drivers.ACTION = cycleDriver;
+      const isSame = {};
+      const getCurrent = store.getState;
+      function stateDriver() {
+        return xs.create({
+          start: listener => {
+            stateListener = listener;
+          },
+          stop: () => {},
+        })
+        .fold((prevState, currState) => {
+          if (prevState === getCurrent) {
+            prevState = getCurrent();
+          }
+          if (prevState === currState) {
+            return isSame;
+          }
+          return currState;
+        }, getCurrent)
```

Comment:
This is to prevent the stream from containing several shallow-equal state objects, which can be the case if an action happens but no part of the state is updated in response.